admin/services/pin.service.js

```diff
@@ -78,4 +80,14 @@ pinService.expireMultiplePins = async (pupilIds) => {
   return pupilDataService.sqlUpdatePinsBatch(data)
 }
 
+/**
+ * Detects possible check form and check window error that should prevent pin generation
+ * @returns {String}
+ */
+pinService.getPinsOverviewError = async () => {
```

Comment:
Isn't this really a wrapper for the validator?   It seems a bit anaemic... perhaps it is not needed?

Reply:
@jon-shipley could just call a validator from the controller and add everything in there indeed.
I just thought of 1) always call a validator from a service 2)keep the validator intact in terms of data service methods and just pass down the required data to-be-validated as we do with the rest of them. If we prefer a direct call to validator that incorporates data service method calls I'll change it.